brave/src/main/java/brave/internal/baggage/BaggageHandler.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.internal.baggage;
+
+import brave.baggage.BaggageField;
+import brave.internal.Nullable;
+import brave.propagation.Propagation;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+import java.util.List;
+
+/**
+ * Handles context storage of one or more baggage fields using one {@link ExtraBaggageFields}
+ * state.
+ *
+ * @param <S> the state that represents one or more baggage fields
+ */
+public interface BaggageHandler<S> {
+  /**
+   * When true, calls to {@link #currentFields(Object)} and {@link #handlesField(BaggageField)}
+   * cannot be cached.
+   */
+  boolean isDynamic();
+
+  /** Don't cache if {@link #isDynamic()}. If not dynamic, the state parameter can be ignored. */
+  List<BaggageField> currentFields(@Nullable S state);
+
+  /** Don't cache if {@link #isDynamic()}. */
+  boolean handlesField(BaggageField field);
+
+  /**
+   * Gets the value of this field in the given state.
+   *
+   * @see BaggageField#getValue(TraceContext)
+   * @see BaggageField#getValue(TraceContextOrSamplingFlags)
+   */
+  @Nullable String getValue(BaggageField field, S state);
+
+  /**
+   * Creates a state object to support the first field value.
+   *
+   * @see BaggageField#updateValue(TraceContext, String)
+   * @see BaggageField#updateValue(TraceContextOrSamplingFlags, String)
+   */
+  S newState(BaggageField field, String value);
+
+  /**
+   * Updates a state object to handle a field value change.
+   *
+   * @see BaggageField#updateValue(TraceContext, String)
+   * @see BaggageField#updateValue(TraceContextOrSamplingFlags, String)
+   */
+  S mergeState(S state, BaggageField field, @Nullable String value);
```

Comment:
Any reason not to use `updateState` like `updateValue`?

Reply:
good!